core/mixins/_retina-bg-image.scss

```diff
@@ -17,11 +17,13 @@
  */
 
 
-@mixin retina-bg-image($image, $width, $height) {
-  @media (-o-min-device-pixel-ratio: 5/4),
-         (-webkit-min-device-pixel-ratio: 1.25),
-         (min-resolution: 120dpi) {
-           background-image: url($image);
-           background-size: $width $height;
+@mixin retina-bg-image($image, $width: auto, $height: auto, $resolution: $retina-resolution) {
+  $width: if($width, $width, auto);
+  $height: if($height, $height, auto);
+  $resolution: if(unitless($resolution), $resolution * 96dpi, $resolution);
+  
+  @media (min-resolution: $resolution) {
+    background-image: url($image);
+    background-size: $width $height;
   }
```

Comment:
Nice one adding `auto` but can we allow it to just take one width value if the height is omitted so that the height will just default to `auto` as per the spec? E.g.

```scss
.foo {
   background: url("logo.png") no-repeat;
   @include retina-bg-image("logo@2x.png", 40px, 3);
 }
```

Would compile to:

```css
.foo {
  background: url("logo.png") no-repeat; 
}
@media (min-resolution: 288dpi) {
   .foo {
     background-image: url("logo@2x.png");
     background-size: 40px auto;
   }
}
```

Also if you want to specify `auto` for both width and height then the mixin makes you apply it twice when you only need to apply it once, [see -> "One-value syntax"](https://developer.mozilla.org/en-US/docs/Web/CSS/background-size#Syntax) but I think you have to do that when working with Sass arguments? Which could also be the case for the 1st thing I mentioned above?